### PR TITLE
Stdm geoodk

### PR DESCRIPTION
**Pull request by SolomonN**
geoodk component is finalized and ready to merge with master.

### CODE REVIEW

**Comment by gkahiu** [comment]
Hi Solomon,

Many thanks for the work on the GeoODK. I have a good look at the desktop component and below are my comments:

1. I propose the title of one of the commands to change from 'Generate Mobile Forms' to 'Generate Mobile Form' in singular form; Should we explicitly mention 'GeoODK' instead of 'Mobile' because the latter is too generalized; it could mean KoboCollect, Magpi, PoiMapper etc.
2. Related to 1. above, change the title of the dialog to GeoODK Form Generator;
3. See the proposed changes in the screenshot below:
comments 05 10 2017 form generator

4. If there is no entity selected, change the message to something like ' Please select at least one entity...'
5. For the check option 'Include Social Tenure Relationship tables', should we rephrase it to 'Enable definition social tenure relationship...' or something that does not explicitly mention tables but something more descriptive.  
6. There are instances where the user checks "Include Social Tenure Relationship Tables' then unchecks either the party or spatial unit entities, the module will still go ahead and export the form. This breaks the first check condition of including STR tables. I suggest you incorporate a check where if a user checks  to include STR tables then unchecks a party(ies) or spatial unit(s) entities, then a warning message should show to indicate that, for example, 'Farm' has been set as the spatial unit in the profile's social tenure relationship but has been excluded from the exported form. Proceed anyway? Click Yes to generate...
7. I still feel there is a gap once the generated ODK file has been exported and the process of saving it to phone, I have this idea of incorporating this as a wizard that provides the option of uploading this to the phone. Let us discuss further, should not be too much work.
8. Please see screenshot below on UI comments for the import module.
comments 05 10 2017 import mobile data

9. I tried to click Save to test validation of an empty selection the got a message 'Current profile matched not records for import...'. Could you rephrase this to 'No records could be found for the current profile?'. Also, the buttons of the dialog end up being disabled afterwards (see highlighted section in the screenshot below), is this by design?
image

These are my quick comments. Am available if you want us to discuss this further.

@pgathogo @wondie any additional comments?

Cheers!

**Comment by wondie** [comment]
Hi Solomon, 

Thank you very much for the great work! I have one question. Can we use Kobo collect with the same tool and workflow? Did you try it with Kobo collect yourself?

**Comment by SolomonN** [comment]
Hello Wondie,

Yes, you can use Kobo Collect. The generated form is standard and would work with any XForm compartible app like ODK. The only challenge with Kobo Collect is that, where you would like to capture a polygon, you cannot do so because it only allow point geometry data.

**Comment by SolomonN** [comment]
90% of the suggestions and comments reviewed and updated. Yet to update the icons in the clickable buttons. Please pull and review.

**Comment by SolomonN** [comment]
90% of the suggestions and comments reviewed and updated. Yet to update the icons in the clickable buttons. Please pull and review.

**Comment by SolomonN** [comment]
90% of the suggestions and comments reviewed and updated. Yet to update the icons in the clickable buttons. Please pull and review.
